iothub/service/design.md

```diff
@@ -0,0 +1,119 @@
+# RBAC API Design for IoT hub
+
+## Service Client
+
+```csharp
+/// <summary>
+/// Create ServiceClient using AAD token credentials and the specified Transport Type.
+/// </summary>
+/// <param name="hostName">IoT hub host name.</param>
+/// <param name="credential">AAD <see cref="TokenCredential"> to authenticate with IoT hub.</param>
+/// <param name="transportType">Specifies whether Amqp or Amqp_WebSocket_Only transport is used.</param>
+/// <param name="transportSettings">Specifies the AMQP and HTTP proxy settings for Service Client.</param>
+/// <param name="options">The options that allow configuration of the service client instance during initialization.</param>
+/// <returns>An instance of <see cref="ServiceClient"/>.</returns>
+public static ServiceClient Create(
+    string hostName,
+    TokenCredential credential,
+    TransportType transportType,
+    ServiceClientTransportSettings transportSettings = default,
+    ServiceClientOptions options = default)
+
+/// <summary>
+/// Create ServiceClient using AAD token credentials and the specified Transport Type.
+/// </summary>
+/// <param name="hostName">IoT hub host name.</param>
+/// <param name="credential">SAS <see cref="AzureSasCredential"> to authenticate with IoT hub.</param>
```

Comment:
do we need to specify "sas" again (it is a part of the type)?

Reply:
I was on the fence on this one. It might be redundant but want to see what others feel.